tuw_nlp/grammar/lexicon.py

```diff
@@ -339,13 +339,21 @@ def handle_acl_relcl(self, parent_pos, current_pos, children_pos, i, j):
         )
         return [acl_relcl, acl_relcl2]
 
-    def handle_obl_case(self, parent_pos, current_pos, children_pos, i):
-        obl_case = (
-            f"{parent_pos} -> handle_obl_case{i}({parent_pos}, {current_pos}, {children_pos})",
-            {'ud': f'f_dep2(merge(merge(?1,"(r<root> :OBL (d2<dep2>))"), r_dep2(f_dep1(merge(merge(?2,"(r<root> :CASE (d1<dep1>))"), r_dep1(?3))))))',
-             'fourlang': f'f_dep2(f_dep1(merge(merge(merge(?1,"(d1<dep1> :1 (r<root>) :2 (d2<dep2>))"), r_dep1(?3)), r_dep2(?2))))'},
-            "nonterminal"
-        )
+    def handle_obl_case(self, parent_pos, current_pos, children_pos, i, clemma):
+        if clemma == "by":
+            obl_case = (
+                f"{parent_pos} -> handle_obl_case{i}({parent_pos}, {current_pos}, {children_pos})",
+                {'ud': f'f_dep2(merge(merge(?1,"(r<root> :OBL (d2<dep2>))"), r_dep2(f_dep1(merge(merge(?2,"(r<root> :CASE (d1<dep1>))"), r_dep1(?3))))))',
+                'fourlang': f'f_dep1(merge(merge(?1,"(r<root> :1 (d1<dep1> :0 (r<root>)))"), r_dep1(?2)))'},
```

Comment:
I'm not sure how this works, but I think the verbatim graph
```
"(r<root> :1 (d1<dep1> :0 (r<root>)))"
```
can't be right, it has two roots...

Reply:
I think that shouldn't be a problem. It handles the graph so in the case of "John is supported by Olaf":
![image](https://user-images.githubusercontent.com/22393845/101155954-2c9cd700-3628-11eb-8276-1ae53e474731.png)

The two root is present because of the edges between support and Olaf (so it goes both way)

Reply:
oh sorry, right, I totally missed it, I thought those were two different nodes. Great!